src/components/Pagination/index.tsx

```diff
@@ -0,0 +1,99 @@
+import {
+  ChevronLeftIcon,
+  ChevronRightIcon,
+  ChevronsLeftIcon,
+  ChevronsRightIcon,
+} from 'lucide-react';
+
+import useSetSearchParams from '@/hooks/useSetSearchParam';
+
+import * as S from './Pagination.css';
+
+interface PaginationProps {
+  totalPage: number;
+  currentPage: number;
+}
+
+const NUM = 5;
+
+function Pagination({ pageInfo }: { pageInfo: PaginationProps }) {
+  const { totalPage, currentPage } = pageInfo;
+  const { setSingleSearchParam } = useSetSearchParams();
+
+  const handlePageChange = (newPage: number) => {
+    setSingleSearchParam('page', newPage.toString());
+  };
+
+  const half = Math.floor(NUM / 2);
+  const startPage = Math.max(1, Math.min(currentPage + 1 - half, totalPage - NUM + 1));
+  const endPage = Math.min(totalPage, startPage + NUM - 1);
+
+  const numberStyle = (page: number) => {
+    return currentPage + 1 === page ? S.base.currentNumbers : S.base.numbers;
```

Comment:
```suggestion
    return currentPage === page ? S.base.currentNumbers : S.base.numbers;
```
`currentPage`이 값을 url에서 가져온 값을 사용한다면 +1이 필요없지 않을까요?